homeassistant/components/fritz/__init__.py

```diff
@@ -20,12 +29,60 @@
     FRITZ_AUTH_EXCEPTIONS,
     FRITZ_EXCEPTIONS,
     PLATFORMS,
+    SERVICE_SET_GUEST_WIFI_PW,
 )
 from .coordinator import AvmWrapper, FritzData
-from .services import async_setup_services, async_unload_services
 
 _LOGGER = logging.getLogger(__name__)
 
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+SERVICE_SCHEMA_SET_GUEST_WIFI_PW = vol.Schema(
+    {
+        vol.Required("device_id"): str,
+        vol.Optional("password"): vol.Length(min=8, max=63),
+        vol.Optional("length"): vol.Range(min=8, max=63),
+    }
+)
+
+SERVICE_LIST: list[tuple[str, vol.Schema | None]] = [
+    (SERVICE_SET_GUEST_WIFI_PW, SERVICE_SCHEMA_SET_GUEST_WIFI_PW),
+]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up fritzboxtools services."""
+    for service, _ in SERVICE_LIST:
+        if hass.services.has_service(DOMAIN, service):
+            return True
```

Comment:
Why are you moving all this into `__init__.py`?
I think it's a lot cleaner when it stays in `services.py`.

Also, async_setup gets run only once, so there is no need to check if the services have already been setup.

Finally, please note that ServiceCall now has a hass property, so you no longer need to have `hass` as an "nested" variable

Reply:
i moved it into `__init__` because "it was not much code" :see_no_evil: but yeah you're right, it's much cleaner when it lives in its own module file :+1: 

thx for the hint about the hass property on ServiceCall - have changed the code accordingly